Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM python:3.11-bullseye
+LABEL org.opencontainers.image.source https://github.com/openzim/great_project
+
+# Install necessary packages
+RUN apt-get update -y \
+    && apt-get install -y --no-install-recommends locales-all \
```

Comment:
This is personnal preference but I like to chain before the break so that every line starts as readable

```dockerfile
RUN apt-get update -y && \
    apt-get install -y --no-install-recommends locales-all && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
```

WDTK? Whichever we choose we should apply it everywhere (progressively)

Reply:
I'd also but the packages grouped on separate lines with additional indentation and a comment on top explaining why recommended.

```dockerfile
RUN apt-get update -y && \
    apt-get install -y --no-install-recommends
        # locales required if tool has any i18n support
        locales-all && \
    apt-get clean
```

Reply:
I agree about packages on separated lines.

Regarding chaining with `&&`, while I get your reasoning, the community seems to be used to the current approach (see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/ for instance) with `&&` at the beginning of the line, so I would prefer to stay it. What can be done is aligning the commands start so that it help a bit in readability.

```dockerfile
RUN apt-get update -y \
 && apt-get install -y --no-install-recommends \
    # locales required if tool has any i18n support
    locales-all \
 && rm -rf /var/lib/apt/lists/* \
 && python -m pip install --no-cache-dir -U pip build
```

Note that I removed `apt-get clean` following the recommendation on the page cited above:
> Official Debian and Ubuntu images [automatically run apt-get clean](https://github.com/moby/moby/blob/03e2923e42446dbb830c654d0eec323a0b4ef02a/contrib/mkimage/debootstrap#L82-L105), so explicit invocation is not required.

Reply:
OK, then. I can also imagine that having the `&&` up front helps understand whether it's a conditional call or not.
Thanks for the research